http/login_test.go

```diff
@@ -2,85 +2,82 @@ package http
 
 import (
 	"bytes"
+	"fmt"
+	"github.com/stretchr/testify/require"
 	"net/http"
 	"net/http/httptest"
-	"strings"
 	"testing"
 )
 
-func TestLogin(t *testing.T) {
-	t.Run("Only GET method allowed", func(t *testing.T) {
-		req, err := http.NewRequest(http.MethodPost, "/", nil)
-		if err != nil {
-			t.Fatalf("could not create request: %v", err)
-		}
-		rec := httptest.NewRecorder()
-		LoginUserHandler(rec, req)
-		if rec.Code != http.StatusMovedPermanently {
-			t.Errorf("expected status %d but got %d", http.StatusMethodNotAllowed, rec.Code)
-		}
-	})
+func TestLoginUserHandler(t *testing.T) {
+	t.Parallel()
+
+	t.Run("Errors", testLoginErrors)
+	t.Run("Success", testLoginSuccess)
+}
+
+func testLoginErrors(t *testing.T) {
+	t.Parallel()
+
+	methodsToTest := []string{http.MethodDelete, http.MethodPost, http.MethodPut, http.MethodPatch, http.MethodOptions}
```

Comment:
Acho que assim ficaria mais fácil de entender. Pense no código como uma leitura, quanto mais claros os nomes das variáveis forem, menos contexto precisa ser carregado na cabeça pra entender o código.

```suggestion
	forbiddenMethods := []string{http.MethodDelete, http.MethodPost, http.MethodPut, http.MethodPatch, http.MethodOptions}
```